weldx/asdf/utils.py

```diff
@@ -95,86 +97,84 @@ def _write_read_buffer(
     return _read_buffer(buffer, open_kwargs)
 
 
-try:  # pragma: no cover
-    import IPython
-    from IPython.display import JSON
+def _get_yaml_header(file) -> str:
+    """Read the YAML header part of an ASDF file.
+
+    Parameters
+    ----------
+    file
+        filename or BytesIO buffer of ASDF file
+
+    Returns
+    -------
+    str
+
+    """
+    if isinstance(file, BytesIO):
+        file.seek(0)
+        code = file.read()
+    else:
+        with open(file, "rb") as f:
+            code = f.read()
+
+    parts = code.partition(b"\n...")
+    code = parts[0].decode("utf-8") + parts[1].decode("utf-8")
+    return code
+
+
+def notebook_fileprinter(file, lexer="YAML"):
+    """Prints the code from file/BytesIO  to notebook cell with syntax highlighting.
+
+    Parameters
+    ----------
+    file
+        filename or BytesIO buffer of ASDF file
+    lexer
+        Syntax style to use
+
+    """
     from pygments import highlight
-    from pygments.formatters import HtmlFormatter
     from pygments.lexers import get_lexer_by_name, get_lexer_for_filename
-except ImportError:  # pragma: no cover
-    pass
-else:  # pragma: no cover
-
-    def _get_yaml_header(file) -> str:
-        """Read the YAML header part of an ASDF file.
-
-        Parameters
-        ----------
-        file
-            filename or BytesIO buffer of ASDF file
-
-        Returns
-        -------
-        str
-
-        """
-        if isinstance(file, BytesIO):
-            file.seek(0)
-            code = file.read()
-        else:
-            with open(file, "rb") as f:
-                code = f.read()
-
-        parts = code.partition(b"\n...")
-        code = parts[0].decode("utf-8") + parts[1].decode("utf-8")
-        return code
-
-    def notebook_fileprinter(file, lexer="YAML"):
-        """Prints the code from file/BytesIO  to notebook cell with syntax highlighting.
-
-        Parameters
-        ----------
-        file
-            filename or BytesIO buffer of ASDF file
-        lexer
-            Syntax style to use
-
-        """
-        if isinstance(file, BytesIO):
-            lexer = get_lexer_by_name(lexer)
-        elif Path(file).suffix == ".asdf":
-            lexer = get_lexer_by_name("YAML")
-        else:
-            lexer = get_lexer_for_filename(file)
-
-        code = _get_yaml_header(file)
-
-        formatter = HtmlFormatter()
-        return IPython.display.HTML(
-            '<style type="text/css">{}</style>{}'.format(
-                formatter.get_style_defs(".highlight"),
-                highlight(code, lexer, formatter),
-            )
+    from pygments.formatters.html import HtmlFormatter
+    from IPython.display import HTML
```

Comment:
The "good" thing about the old implementation wrapped in 

```python
try:
...
except ImportError:
pass
```
was that the function would simply not get defined in the wrong context (without IPython). But running all imports is a bit heavy I agree.
Maybe we can use a construct like
```python
if importlib.util.find_spec("IPython") and importlib.util.find_spec("pygments"):
def notebook_fileprinter(file, lexer="YAML"):
...
```

Reply:
So find_spec does not do the heavy import, but only checks if the package is available?

Reply:
yes it should simply return `None` if the package is not installed (and some package metadata if it is) 
https://docs.python.org/3/library/importlib.html#importlib.util.find_spec

Reply:
Having functions available only if a certain package is available/importable seems kind of confusing. You would just get an ImportError/AttributeError further downstream, if IPython is not in place. I suggest, we define these tools as stubs if Ipython is not there, and define/implement them for the other case.

Reply:
or just keep your current proposed implementation, calling these functions really doesn't make any sense outside of an IPython of Juypter context

Reply:
Now we have stubs, if pygments/ipython is not available and proper implementation for the usual case.